### DISCUSSION TITLE
fully routing an android app problem

### DISCUSSION THREAD

**Discussion by sharif221** [XTLS/Xray-core#3764] [Q&A] [answered]
In this scenario
where there is one X-ray server and one Android client connected to the X-ray server.
in the x-ray, I have 2 outbound:

freedom
custom outbound connected to socks5 local proxy

and I have a custom rule:

```  
{
    "type": "field",
    "domain": [
      "geosite:spotify",
      "scdn.co",
      "spotify.com",
      "spotifycdn.com"
    ],
    "outboundTag": "WARP"
  },
```
it does not catch the Spotify I still get not connected error or something like that
but when I fully use the WARP and forwarding all traffic to it it suddenly get fixed.
also its only for app and website id fully functioning 

I don't know why this happens I'm certain that I fully catched all spotify domain using network tools, is there any setting that I can force not only domain but also ips that the domain have to be catched?

**Comment by mmmray** [comment, marked as answer]
https://github.com/XTLS/Xray-core/issues/3289#issuecomment-2208976955

**Reply by sharif221** [reply]
that's right thanks for response 👍
